### PR TITLE
Feature/more colormaps

### PR DESCRIPTION
**Pull request by nathanhhughes**
@Schmluk Moves colormaps to new file and namespace and adds a couple discrete palettes that we talked about

### CODE REVIEW

File: include/spark_dsg/colormaps.h [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * @param start The HLS color corresponding to 0
+ * @param end The HLS color corresponding to 1
+ */
+Color hls(float value,
+          const std::array<float, 3>& start,
+          const std::array<float, 3>& end);
+
+/**
+ * @brief Generate a color from a general divergent colormap
+ * @param value A value in [0, 1]. 0 results in one color, 1 in the other, and 0.5 in a
```
Comment (Schmluk) [inline include/spark_dsg/colormaps.h:97]: Minor: The docstring is not very informative, maybe say 0 results in the low-hue color, 1 in the high-hue color?

File: src/colormaps.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
+#include "spark_dsg/colormaps.h"
+
+#include <algorithm>
+#include <random>
+
+namespace spark_dsg::colormaps {
+
+namespace {
```
Comment (Schmluk) [inline src/colormaps.cpp:8]: Is there a reason this is in anonymous namespace? Seems like standard helper functions to me.

**Review by Schmluk — APPROVED** [review]
Nice! LGTM

File: src/colormaps.cpp (hunk 2 above)
Comment (nathanhhughes) [reply src/colormaps.cpp:8]: I'm trying to be a little better about avoiding odr issues (even if it's unlikely someone else is going to define a `spark_dsg::colormaps::exponentialOffsetId` function elsewhere in the code. Happy to add a declaration in the header and expose it publicly though if you think it's useful downstream independent of `rainbowId`

File: include/spark_dsg/colormaps.h (hunk 1 above)
Comment (nathanhhughes) [reply include/spark_dsg/colormaps.h:97]: I would guess I was just trying avoid a line break, but agreed :sweat_smile:

**Pull request merged**